cedar-team/django-bulk-load
# feat: select_for_update in bulk_select_model_dicts

## Summary
Ticket: https://cedar-jira.atlassian.net/browse/CED-67107

This adds the ability to add `FOR UPDATE` to the `bulk_select_model_dicts` function. This is useful for locking to rows when performing an update

## Test Plan
##### How can reviewers test your change manually?
Check changed tests. Call the function with the new argument and check the rows are locked
##### What automated tests did you add? If none, please state why.
Added automated tests to check for lock

## Security Impact
Please answer the questions below about your PR. 
If yes, please explain and add `cedar-team/security` as reviewers.

##### Add/modify authentication, authorization, encryption, or HTTP headers?
No
##### Add/modify any PII or PHI data fields?
No
##### Add new third parties the app interacts with (e.g., Nordis/Stripe)?
No
##### Ask patients to enter a new kind of data, (e.g. insurance capture)?
No
##### Impact any of the other [Security Engineering Practices](https://careportal.atlassian.net/wiki/spaces/SEC/pages/1192690295/Security+Engineering+Practices)?
_Protect sensitive data, Validate your inputs, Know your user, Establish trust boundaries, Keep good records, Expect vulnerabilities, Follow least privilege_
No
